python/cuml/common/array.py

```diff
@@ -112,15 +112,19 @@ def __init__(self, data=None, owner=None, dtype=None, shape=None,
         if data is None:
             raise TypeError("To create an empty Array, use the class method" +
                             " Array.empty()")
-        if isinstance(data, int):
+        if _check_low_level_type(data):
             if dtype is None or shape is None or order is None:
                 raise TypeError("Need to specify dtype, shape and order when" +
                                 " creating an Array from a pointer.")
+            detailed_construction = True
+        else:
+            detailed_construction = False
 
         ary_interface = False
 
-        if isinstance(data, DeviceBuffer) or isinstance(data, int) or \
-                isinstance(data, Buffer):
+        if detailed_construction:
```

Comment:
should probably update the class header doc to say data can be a serialized bytearray